gossip-base/src/main/java/org/apache/gossip/manager/GossipCore.java

```diff
@@ -55,13 +61,16 @@
   private final Meter messageSerdeException;
   private final Meter tranmissionException;
   private final Meter tranmissionSuccess;
-
+  private final List<UpdateNodeDataEventHandler> nodeDataHandlers;
+  private final List<UpdateSharedDataEventHandler> sharedDataHandlers;
   public GossipCore(GossipManager manager, MetricRegistry metrics){
     this.gossipManager = manager;
     requests = new ConcurrentHashMap<>();
     workQueue = new ArrayBlockingQueue<>(1024);
     perNodeData = new ConcurrentHashMap<>();
     sharedData = new ConcurrentHashMap<>();
+    nodeDataHandlers = new ArrayList<>();   // TODO: 5/30/17 check for concurrency issues
```

Comment:
I am not sure either. I would say use a CopyOnWriteArrayList or some concurrent structure from java.util.concurrent